server/protocols/jmap/src/test/java/org/apache/james/jmap/mailet/filter/JMAPFilteringTest.java

```diff
@@ -0,0 +1,3393 @@
+                        ImmutableList.of(
+                            Rule.builder()
+                                .id(RULE_ID_1)
+                                .name(RULE_NAME_1)
+                                .condition(Rule.Condition.of(Rule.Condition.Field.FROM, Rule.Condition.Comparator.CONTAINS, SENDER_1_USERNAME))
+                                .action(Rule.Action.of(Rule.Action.AppendInMailboxes.withMailboxIds(ImmutableList.of(testSystem.getRecipient1MailboxId().serialize()))))
+                                .build()));
+
+                FakeMail mail = FakeMail.builder()
+                        .sender(SENDER_1)
+                        .recipients(RECIPIENT_1, RECIPIENT_2)
+                        .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                            .addHeader("from", "sender2 <sender2@james.org>, lina <lina@james.org>"))
+                        .build();
+
+                testSystem.getJmapFiltering().service(mail);
+
+                assertThat(mail.getAttribute(DELIVERY_PATH_PREFIX + RECIPIENT_1_USERNAME))
+                        .isNull();
+            }
+
+            @Test
+            @Override
+            public void mailDirectiveShouldBeSetWhenAtLeastOneContainsRuleValue(JMAPFilteringTestSystem testSystem) throws Exception {
+                testSystem.getFilteringManagement().defineRulesForUser(User.fromUsername(RECIPIENT_1_USERNAME),
+                        ImmutableList.of(
+                            Rule.builder()
+                                .id(RULE_ID_1)
+                                .name(RULE_NAME_1)
+                                .condition(Rule.Condition.of(Rule.Condition.Field.FROM, Rule.Condition.Comparator.CONTAINS, SENDER_1_USERNAME))
+                                .action(Rule.Action.of(Rule.Action.AppendInMailboxes.withMailboxIds(ImmutableList.of(testSystem.getRecipient1MailboxId().serialize()))))
+                                .build()));
+
+                FakeMail mail = FakeMail.builder()
+                        .sender(SENDER_1)
+                        .recipients(RECIPIENT_1, RECIPIENT_2)
+                        .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                            .addHeader("from", "sender1 <sender1@james.org>, lina <lina@james.org>"))
+                        .build();
+
+                testSystem.getJmapFiltering().service(mail);
+
+                assertThat(mail.getAttribute(DELIVERY_PATH_PREFIX + RECIPIENT_1_USERNAME))
+                        .isEqualTo(RECIPIENT_1_MAILBOX_1);
+            }
+        }
+
+        @Nested
+        class FromFilteringWithNotContainsCommandRule implements WithNotContainsCommandRule {
+
+            @Test
+            @Override
+            public void mailDirectiveShouldBeSetWhenDoesntContainsRuleValue(JMAPFilteringTestSystem testSystem) throws Exception {
+                testSystem.getFilteringManagement().defineRulesForUser(User.fromUsername(RECIPIENT_1_USERNAME),
```

Comment:
Your test system is based here on a single rule. IMO you can make it a bit less verbose by adding the following method to your testSystem: `setRule(User, Condition, Action)`

It will reduce a bit the boiler plate.

BTW `defineRulesForUser` can also be called without the immutableList